### PR TITLE
Promote WatchList feature to Beta

### PR DESCRIPTION
**Pull request by p0lyn0mial**
#### What type of PR is this?
/kind feature

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
Graduates the `WatchList` feature gate to Beta for kube-apiserver and enables `WatchListClient` for KCM.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: cmd/kube-controller-manager/app/options/options.go [hunk 1]
```diff
@@ -273,6 +276,13 @@ func (s *KubeControllerManagerOptions) Flags(allControllers []string, disabledBy
 	fs := fss.FlagSet("misc")
 	fs.StringVar(&s.Master, "master", s.Master, "The address of the Kubernetes API server (overrides any value in kubeconfig).")
 	fs.StringVar(&s.Generic.ClientConnection.Kubeconfig, "kubeconfig", s.Generic.ClientConnection.Kubeconfig, "Path to kubeconfig file with authorization and master location information (the master location can be overridden by the master flag).")
+
+	if !utilfeature.DefaultFeatureGate.Enabled(featuregate.Feature(clientgofeaturegate.WatchListClient)) {
+		if err := utilfeature.DefaultMutableFeatureGate.OverrideDefault(featuregate.Feature(clientgofeaturegate.WatchListClient), true); err != nil {
```
Comment (wojtek-t) [inline cmd/kube-controller-manager/app/options/options.go:281]: nit: add a comment that this error will currently be logged in integration tests and why

File: cmd/kube-controller-manager/app/options/options_test.go [hunk 2]
```diff
@@ -1320,50 +1320,54 @@ func TestControllerManagerAliases(t *testing.T) {
...
-	if fg == nil {
-		t.Fatalf("didn't find %q flag", fgFlagName)
-	}
-
-	expectedWatchListClientString := "WatchListClient=true|false (BETA - default=false)"
-	if !strings.Contains(fg.Usage, expectedWatchListClientString) {
-		t.Fatalf("%q flag doesn't contain the expected usage for %v feature gate.\nExpected = %v\nUsage = %v", fgFlagName, clientgofeaturegate.WatchListClient, expectedWatchListClientString, fg.Usage)
-	}
+	assertWatchListClientFeatureDefaultValue(t)
+	assertWatchListCommandLineDefaultValue(t, fs)
```
Comment (wojtek-t) [inline cmd/kube-controller-manager/app/options/options_test.go:1330]: Can you move this to a separate PR, at least to some extent it seems like a cleanup.

Comment (p0lyn0mial) [reply cmd/kube-controller-manager/app/options/options_test.go:1330]: sure, opened https://github.com/kubernetes/kubernetes/pull/125723

**Comment by wojtek-t** [comment]
Based on https://github.com/kubernetes/kubernetes/issues/125760#issuecomment-2199520596 and the fact the test is not even using watchcache, this certainly doesn't seem related.

/lgtm
/approve
